apps/hash-ai-worker-ts/src/activities/infer-entities-from-web-page-activity.ts

```diff
@@ -84,12 +84,19 @@ export const inferEntitiesFromWebPageActivity = async (params: {
     ${typeof webPage === "string" ? "The content of the web page is as follows:" : `The website page title is ${webPage.title}, hosted at ${webPage.url}. Its content is as follows:`}
     ${typeof webPage === "string" ? webPage : webPage.textContent}
     ---WEBSITE CONTENT ENDS---
-    
-    You already provided a summary of the ${
-      relevantEntitiesPrompt
-        ? "relevant entities you inferred"
-        : "entities you can infer"
-    } from the website. Here it is:
+
+    If the content appears to include the data from a table, you must
+      apply any necessary unit conversions from the table header so
+      that values stored on entity properties are correct.
+
+    Pay careful attention to the units of data, which may be defined
+      in a column header, or in the table row.
+
+    If you are asked to store a unit alongside a value in properties,
+      ensure that the values is stored in the correct unit. Do not
+      change units, you must use the units specified in the data.
```

Comment:
I find this a little confusing so I assume the LLM will too – it says both that it must "apply any necessary unit conversions" but also "not change units". What are the 'necessary unit conversions' it should perform, if it's using the unit from the data?

Reply:
Good spot, at one point I attempted to get it to convert the units and at another I wanted to discourage it. I've removed the conversion related sentence in https://github.com/hashintel/hash/pull/4305/commits/ba8fbda5602a0b332db07b470fe630c2393267d0, once I spend more time on the inference agent I expect this prompt will be reworked further.